scripts/biosamples_relationships/src/biosamples_relationships.py

```diff
@@ -224,38 +245,82 @@ def add_relationships(file_data, source_bs, target_list, r_type = "derived from"
+        f.write(r.text)  # Saves sample json object to file
+
     file_data = json.loads(r.text)
-    file_data.pop("_links") # Removes links array (will be added automatically after updating the biosample)
+    file_data.pop("_links")  # Removes links array (will be added automatically after updating the biosample)
 
     # We call the function that returns the list of relationships
-    rel_list = add_relationships(file_data = file_data, 
-                                 source_bs = source_bs, 
-                                 target_list = df_dict[source_bs])
+    rel_list = relationships(file_data=file_data,
+                             source_bs=source_bs,
+                             target_list=df_dict[source_bs])
 
-    # We overwrite the relationships node in our array and file
-    rel_array = {"relationships": rel_list}
-    file_data.update(rel_array)
+    target_bs = new_relationship['target'] #setting target_bs variable here
 
-    edited_source_file = os.path.join(output_dir, f"linked_{source_bs}.json")
-    with open(edited_source_file, 'w') as f:
-        json.dump(file_data, f, indent = 1) # Converts python dictionary back into json string
+    # Editing SOURCE/ENA sample metadata to link samples:
+    if not args.unlink:
+        # We first append the new relationship to the list (only if it's not already there)
+        if new_relationship in rel_list:
+            print_v(f"- WARNING: Relationship of source sample '{source_bs}' derived from target sample '{target_bs}' already existed in the downloaded JSON file. Skipping.") #notdefined- why?
+            continue
+        rel_list.append(new_relationship)
 
-# Submit updated Biosample json file:    
-    update_url = "{0}{1}".format(biosamples_url, webin_auth)  
+        # We overwrite the relationships node in our array and file
+        rel_array = {"relationships": rel_list}
+        file_data.update(rel_array)
+
+        # We update the JSON file with the new relationships
+        edited_source_file = os.path.join(output_dir, f"linked_{source_bs}.json")
+        with open(edited_source_file, 'w') as f:
+            json.dump(file_data, f, indent=1)  # Converts python dictionary back into json string
+
+    # Editing SOURCE/ENA sample metadata to unlink samples:
+    elif args.unlink:
+        #interactive check whether user would really like to remove unlink samples
+        user_response = input(f" - \n WARNING: Relationship between source sample '{source_bs}' derived from target sample '{target_bs}' will be removed. Do you wish to proceed? [y/n]")
+        if not user_response == "y":
+            print("\t Aborting script.")
+            sys.exit()
+        # Remove relationships between source + target accs
+        try:
+            index = file_data["relationships"].index(new_relationship)
+            file_data["relationships"].pop(index) # removing nested lists using integers not strings
+        except: #except if ValueError thrown
```

Comment:
I would suggest that, if we know already the exception, we catch it appropriately. 
```suggestion
        except ValueError: #except if ValueError thrown
```